### ISSUE TITLE
api: extension-api-client throws `Error invoking remote method` error if returned HTTP code is not StatusOK

### ISSUE THREAD

**Issue by Dentrax** [docker/extensions-sdk#188]
After so loooooooooooong time of debugging, I finally figured it out why I'm getting the following error:

```
Uncaught (in promise) Error: Error invoking remote method 'furkan_hello-docker-extension-backend-request': [object Object]
```

That's because I was returning `http.StatusInternalServerError`. But... `@docker/extension-api-client` only works as expected in cases we return `http.StatusOK`.

Works:
```golang
func hello(ctx echo.Context) error {
	return ctx.JSON(http.StatusOK, HTTPMessageBody{Message: "hello"})
}
```

Does not works:
```golang
func hello(ctx echo.Context) error {
	return ctx.JSON(http.StatusInternalServerError, HTTPMessageBody{Message: "hello"})
}
```

My expected case is that extension-api-client should **successfully handle** all request/response **regardless of** the HTTP **status code**.

Versions:
```
Extensions SDK
Client Version: v0.2.4
Server API Version: 0.2.3

Docker Desktop
Version: 4.8.2 (79419) (macOS Intel)
Engine: 20.10.14
Compose: 1.29.2
Credential Helper: v0.6.4
Kubernetes: v1.24.0
Snyk: v1.827.0
```

**Comment by Dentrax** [comment]
Sorry, forgot to add an example function of the SDK:

```ts
const result = await ddClient.extension.vm.service.get('/hello');
```

Return a non-200 response:

```golang
func hello(ctx echo.Context) error {
	return ctx.JSON(http.StatusInternalServerError, HTTPMessageBody{Message: "hello"})
}
```

Expect the same error.

**Comment by benja-M-1** [comment]
I managed to reproduce thanks @Dentrax

**Comment by benja-M-1** [comment]
@Dentrax a fix for that will be available in 4.10.0 that is around the corner ;)

**Comment by gtardif** [comment]
This has been shipped now, closing this (feel free to re-open if there are still issues getting the error)
